components/automate-ui/src/app/page-components/simple-line-graph/simple-line-graph.component.ts

```diff
@@ -0,0 +1,342 @@
+import {
+  Component, Input, OnChanges, ViewChild, ElementRef, SimpleChanges
+} from '@angular/core';
+import * as d3 from 'd3';
+import {
+  DayPercentage
+} from 'app/entities/desktop/desktop.model';
+
+
+@Component({
+  selector: 'app-simple-line-graph',
+  templateUrl: './simple-line-graph.component.html',
+  styleUrls: ['./simple-line-graph.component.scss']
+})
+
+export class SimpleLineGraphComponent implements OnChanges {
+
+  constructor(
+    private chart: ElementRef
+  ) {}
+
+  @ViewChild('svg', {static: true}) svg: ElementRef;
+  @Input() data: DayPercentage[] = [];
+  @Input() width = 900;
+  @Input() height = 156; // we want a 116px height on the ticks, this minus margins
+  private locked: number = null;
```

Comment:
`locked` stores a number to reference when the graph is being redrawn or resized, so that it can re-attach to whatever the selected number was.